analyzer/src/main/kotlin/MavenSupport.kt

```diff
@@ -135,7 +136,15 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
                     val type = it.group("type")
                     val url = it.group("url")
 
+                    val userHostMatcher = USER_HOST_REGEX.matcher(type)
+
                     when {
+                        // Some projects omit the type fragment and use the SCP-like Git URL syntax.
+                        userHostMatcher.matches() -> {
+                            val host = userHostMatcher.group("host")
+                            VcsInfo(type = VcsType.GIT, url = "https://$host/$url", revision = tag)
+                        }
+
```

Comment:
IMO this whole logic should go into the `else` block that starts at original line 166, as we already handle invalid `scm` tags that nevertheless clearly are Git URL over there.

Reply:
Sorry, I overlooked that the URLs you'd like to support actually *do* have the "scm:" prefix. But how about

                        type == "git@github.com" -> {
                            VcsInfo(
                                type = VcsType.GIT_REPO,
                                url = "https://github.com/$url",
                                revision = tag
                            )
                        }

even if that's less generic but therefore more explicit?

Reply:
Yet another alternative: We also disallow `@` in the `scm`'s `type` so the `else` from line 166 *is* taken, and handle this case there together with the other invalid cases.

Reply:
I do not understand why the alternative implementations you propose are better and what issue you see with the current implementation respectively.

Reply:
It's not really a technical issues I'm seeing, it's rather an "organizational" one: I'd prefer to handle all cases for correct syntax in one place (inside the `when`), and the cases for incorrect syntax in another place (inside the outer `else`), instead of mixing these code paths.

Reply:
Ok, done.